lucet-runtime/lucet-runtime-internals/src/module/dl.rs

```diff
@@ -65,6 +79,12 @@ impl DlModule {
             unsafe { slice::from_raw_parts(*module_data_ptr, *module_data_len) };
         let module_data = ModuleData::deserialize(module_data_slice)?;
 
+        // If a public key has been provided, verify the module signature
+        // The TOCTOU issue is unavoidable without reimplenting `dlopen(3)`
```

Comment:
after spending some time thinking about this, my only conclusion is `"dlopen() not taking a file descriptor is a crime"`

Reply:
🤷